runtime/doc/lsp.txt

```diff
@@ -1469,6 +1469,42 @@ save({lenses}, {bufnr}, {client_id})                 *vim.lsp.codelens.save()*
       • {client_id}  (integer)
 
 
+==============================================================================
+Lua module: vim.lsp.inlay_hint                                *lsp-inlay_hint*
```

Comment:
Should we move `vim.lsp.inlay_hint()` to this module as well (maybe renaming it to `toggle`)?

Reply:
I'm a fan of `toggle`, I initially suggested that before I figured out how to overload it in the docs. I still want to do it since I'm not a huge fan of using
<img width="621" alt="Screenshot 2023-10-09 at 6 01 11 PM" src="https://github.com/neovim/neovim/assets/5601392/8aec6457-a7be-46de-a06b-b2799fb04af9">
 callable objects for APIs

That being said, I'm probably the most cavalier person when it comes to deprecations and/or breaking changes so I'll defer to others on this.

One somewhat awkward thing is that cmp displays the completion for the function variant (which would be deprecated) until you add the dot after it, so users might not realize that there's non-deprecated stuff after the dot. Although putting the deprecation message on a newline seems like a pretty good fix:

Reply:
Hmm the documentation from the completion entry makes me think that LuaLS is returning the results from both the module definition and the current function, which does feel confusing (and potential bug farm IMO).

Reply:
Actually, YOLO, why be timid.

Including the change in the PR for now until further advised. Inlay hints never released in stable (0.9) anyway

Reply:
I support this decision, so at least you won't be alone in the upcoming (?) debate 😉

Reply:
It’s not considered a breaking change until it’s included in a release, so changing this is fine as long as there’s (technical) agreement.

Reply:
removed